src/components/ChannelsAvailabilityCard/utils.ts

```diff
@@ -0,0 +1,95 @@
+import { ChannelData } from "@saleor/channels/utils";
+import { LocalizeDate } from "@saleor/hooks/useDateLocalize";
+import { IntlShape } from "react-intl";
+
+import { Message } from "./types";
+
+export const getChannelsAvailabilityMessages = ({
```

Comment:
I'm not convinced that we gain anything by using this util
@jwm0 ?

Reply:
Look on master at `src/components/AvailabilityCard/AvailabilityCard.tsx`, the only thing this component does is wrapping another component with those messages. So IMO messages selection is a job for a util function

Reply:
I'm not a fan of either approach. Parent does not need to know/keep all child translations. I'd recommend exposing these messages as a single `messages.ts` file and having each `Channel` pick the correct translation

```
// Channel component

/// current
<Typography variant="caption">
  {messages.availableDateText}
</Typography>

// proposed
getAvailableDateText(publicationDate, isPublished): string

<Typography variant="caption">
  {getAvailableDateText(publicationDate, isPublished)}
</Typography>
```